### DISCUSSION TITLE
Getting queried items in exact iteration order

### DISCUSSION THREAD

**Discussion by luciandavidescu** [directus/directus#11283] [Archive]
Is there a way to return an array of items in the exact iteration order instead of the default?

For example, if the database holds the items _1,2,3_ in that respective order, I'd like to be able to get them as _3,1,2_.
With GraphQL...

Query:

`{items(filter: {id: {_in: ["3","1","2"] } } ) {data}}`

Expected result:

`{"data": {"items": [{"data": "data-from-3"},{"data": "data-from-1"},{"data": "data-from-1"}]}}`

Actual result:

`{"data": {"items": [{"data": "data-from-1"},{"data": "data-from-2"},{"data": "data-from-3"}]}}`

I figure there might be something along the lines of `sort:["_in"]` or `orderby:{operator:"_in"}`

For now, I went with prebuilding the query into something like this...

```
{ 
_3: items(filter:{id: { _eq: "3" }}){data}
_1: items(filter:{id: { _eq: "1" }}){data}
_2: items(filter:{id: { _eq: "2" }}){data}
}
```
...which does work, but I'd rather be sure I'm not doing trivial things in an awqward way.

Use case: Lists built outside the DB, such as popular/trending items (from social/analytics apis) or related items (by algorythm).

**Comment by rijkvanzanten** [comment]
Heya! There's no way to sort by a pre-defined list from the filter query. That being said, you could use the `sort` query parameter to sort by any of the other fields. This allows you to retrieve the items in a manually sorted order (eg through a `sort` field) or by any other value (like number of upvotes for popular, etc) 👍🏻

**Reply by luciandavidescu** [reply]
indeed, some info could be stored in the DB itself first, but for what's generated on the fly or already at hand that round trip would make no sense. going with the aliases, they do get the job done.
